### ISSUE TITLE
`randomITensor` can throw OOM error

### ISSUE THREAD

**Issue by nickrobinson251** [ITensor/ITensors.jl#6]
```julia
julia> using ITensors

julia> randomITensor(2)
julia-1.1(94967,0x7fffb6654380) malloc: *** mach_vm_map(size=811454977087225856) failed (error code=3)
*** error: can't allocate region
*** set a breakpoint in malloc_error_break to debug
ERROR: OutOfMemoryError()
Stacktrace:
 [1] Type at ./boot.jl:402 [inlined]
 [2] Type at ./boot.jl:411 [inlined]
 [3] fill at ./array.jl:403 [inlined]
 [4] fill at ./array.jl:401 [inlined]
 [5] Type at /Users/nick/juliacon/ITensors.jl/src/storage/dense.jl:9 [inlined]
 [6] Type at /Users/nick/juliacon/ITensors.jl/src/itensor.jl:31 [inlined]
 [7] randomITensor at /Users/nick/juliacon/ITensors.jl/src/itensor.jl:187 [inlined]
 [8] randomITensor(::Int64) at /Users/nick/juliacon/ITensors.jl/src/itensor.jl:192
 [9] top-level scope at none:0
```
```
julia> VERSION
v"1.1.1"
```

**Comment by nickrobinson251** [comment]
it's not Int specific 
```julia
julia> using ITensors

julia> randomITensor(IndexSet(2))
julia-1.1(1118,0x7fffb6654380) malloc: *** mach_vm_map(size=1497571080875364352) failed (error code=3)
*** error: can't allocate region
*** set a breakpoint in malloc_error_break to debug
ERROR: OutOfMemoryError()
Stacktrace:
 [1] Type at ./boot.jl:402 [inlined]
 [2] Type at ./boot.jl:411 [inlined]
 [3] fill at ./array.jl:403 [inlined]
 [4] fill at ./array.jl:401 [inlined]
 [5] Type at /Users/nick/juliacon/ITensors.jl/src/storage/dense.jl:9 [inlined]
 [6] Type at /Users/nick/juliacon/ITensors.jl/src/itensor.jl:31 [inlined]
 [7] randomITensor at /Users/nick/juliacon/ITensors.jl/src/itensor.jl:187 [inlined]
 [8] randomITensor(::IndexSet) at /Users/nick/juliacon/ITensors.jl/src/itensor.jl:192
 [9] top-level scope at none:0
```
these both hti the `src/itensor.jl:193` method (on current `master`): `randomITensor(inds) = randomITensor(Float64,IndexSet(inds))`

**Comment by YiqingZhouKelly** [comment]
Hi,

This is not an official answer but I hope this will help. :) 

I guess you are trying to construct a random vector with 2 elements. Here's one way to do so:

```Julia
julia> i = Index(2) # construct Index object, index i runs from 1 to 2
(2|id=34)

julia> v = randomITensor(i)
ITensor ord=1 (2|id=34)
Dense{Float64}

julia> print(v)
ITensor ord=1 (2|id=34)
Dense{Float64}
 -0.24215665883263277
 -0.5959806592024572 

julia> j = Index(2)
(2|id=22)

julia> A = randomITensor(i,j) # construct a 2x2 matrix
ITensor ord=2 (2|id=34) (2|id=22)
Dense{Float64}

julia> set = IndexSet(i,j) # or if you want to use IndexSet
IndexSet(Index[(2|id=34), (2|id=22)])

julia> B = randomITensor(set)
ITensor ord=2 (2|id=34) (2|id=22)
Dense{Float64}

julia> print(B)
ITensor ord=2 (2|id=34) (2|id=22)
Dense{Float64}
 0.32607127987730933  -0.6800154984844948
 0.2190657680428048    1.4148377634389848
```

**Comment by mtfishman** [comment]
Thanks for pointing this out. That constructor should limit what it accepts better. The point was to allow the constructor to accept either `IndexSet` or `NTuple{N,Index}`, or anything else that may be "convertible" to IndexSet, but that causes problems since IndexSet accepts an Int in one of it's constructors which makes an uninitialized IndexSet that causes problems later on.

I think for now we can just add the signature `randomITensor(inds::Union{IndexSet,NTuple{N,Index}})` to catch this.

**Comment by mtfishman** [comment]
And also see Yiqing's answer for how this constructor is supposed to be used. It should also accept inputs like:
```julia
randomITensor((i,j))
```
but maybe that one isn't totally necessary and we can see if it is useful.

**Comment by emstoudenmire** [comment]
Fixed in the above commit d9a1478
